src/Shared/SignalVersionHelper.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Reflection;
+
+namespace OpenTelemetry.Instrumentation;
+
+internal static class SignalVersionHelper
+{
+    public static string GetVersion<T>()
+    {
+        return typeof(T).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()!.InformationalVersion.Split('+')[0];
```

Comment:
Since we are using `!.` here, would it be better to use any fallback, or maybe a `?.` chain?

Reply:
I do not see any case when the null value can happen in any of library.
As long as we are using MinVer'sioning it should be safe to use this construction.